sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -582,7 +582,12 @@ def check(self, _):
             if self.autodiscovery and self.autodiscovery_db_service_check:
                 for db_name in self.databases:
                     if db_name != self.connection.DEFAULT_DATABASE:
-                        self.connection.check_database_conns(db_name)
+                        try:
+                            self.connection.check_database_conns(db_name)
+                        except Exception as e:
+                            # service_check errors on auto discovered databases should not abort the check
+                            self.log.debug("failed service check for auto discovered database: %s", e)
```

Comment:
Should this be a warning to make it visible in the logs? It shouldn't be a very high volume error log so it's probably good to include by default without having to enable debug logging.